File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/browsing/library/access/session/SelectedBrowserLibraryIdentifierProvider.java
package com.lasthopesoftware.bluewater.client.browsing.library.access.session;

import android.content.Context;
import android.preference.PreferenceManager;

import com.lasthopesoftware.bluewater.client.browsing.library.repository.LibraryId;

/**
 * Created by david on 2/12/17.
 */
public class SelectedBrowserLibraryIdentifierProvider implements ISelectedLibraryIdentifierProvider {

	private final Context context;

	public SelectedBrowserLibraryIdentifierProvider(Context context) {
		this.context = context;
	}

	@Override
	public LibraryId getSelectedLibraryId() {
		final int libraryId = PreferenceManager.getDefaultSharedPreferences(context).getInt(LibrarySelectionKey.chosenLibraryKey, -1);
		return libraryId > -1 ? new LibraryId(libraryId) : null;
	}
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/browsing/library/access/session/SelectedBrowserLibraryIdentifierProvider.kt
package com.lasthopesoftware.bluewater.client.browsing.library.access.session

import com.lasthopesoftware.bluewater.client.browsing.library.repository.LibraryId
import com.lasthopesoftware.bluewater.settings.repository.ApplicationSettings
import com.lasthopesoftware.bluewater.settings.repository.access.HoldApplicationSettings
import com.namehillsoftware.handoff.promises.Promise
import com.namehillsoftware.handoff.promises.response.ImmediateResponse

class SelectedBrowserLibraryIdentifierProvider(private val applicationSettings: HoldApplicationSettings)
	: ProvideSelectedLibraryId, ImmediateResponse<ApplicationSettings, LibraryId?>
{
	override val selectedLibraryId: Promise<LibraryId?>
		get() = applicationSettings.promiseApplicationSettings().then(this)

	override fun respond(settings: ApplicationSettings): LibraryId? =
		settings.chosenLibraryId.takeIf { it > -1 }?.let(::LibraryId)
}
